library/Cube/Web/Controller.php

```diff
@@ -4,98 +4,156 @@
 
 namespace Icinga\Module\Cube\Web;
 
-use Icinga\Application\Modules\Module;
+use Icinga\Module\Cube\Cube;
 use Icinga\Module\Cube\DimensionParams;
 use Icinga\Module\Cube\Forms\DimensionsForm;
 use Icinga\Module\Cube\Hook\IcingaDbActionsHook;
-use Icinga\Module\Cube\ProvidedHook\Icingadb\IcingadbSupport;
-use Icinga\Web\Controller as WebController;
+use Icinga\Module\Cube\IcingaDb\CustomVariableDimension;
+use Icinga\Module\Cube\IcingaDb\IcingaDbCube;
+use Icinga\Module\Icingadb\Common\Auth;
+use Icinga\Module\Icingadb\Common\Database;
+use Icinga\Module\Icingadb\Web\Control\ProblemToggle;
 use Icinga\Web\View;
+use Icinga\Web\Widget\Tabextension\DashboardAction;
+use ipl\Html\HtmlString;
+use ipl\Orm\Query;
+use ipl\Orm\UnionQuery;
+use ipl\Stdlib\Filter;
+use ipl\Web\Compat\CompatController;
+use ipl\Web\Compat\SearchControls;
+use ipl\Web\Filter\QueryString;
+use ipl\Web\Url;
+use ipl\Web\Widget\Tabs;
 
-abstract class Controller extends WebController
+abstract class Controller extends CompatController
 {
+    use SearchControls;
+    use Database;
+    use Auth;
+
     /** @var View This helps IDEs to understand that this is not ZF view */
     public $view;
 
-    /** @var \Icinga\Module\Cube\Cube */
+    /** @var IcingaDbCube */
     protected $cube;
 
+    /** @var string[] Preserved params for searchbar and search editor controls */
+    protected $preserveParams = [
+        'dimensions',
+        'showSettings',
+        'wantNull',
+        'problems',
+        'sort'
+    ];
+
+    /** @var Filter\Rule Filter from query string parameters */
+    private $filter;
+
     /**
      * Return this controllers' cube
      *
-     * @return \Icinga\Module\Cube\Cube
+     * @return IcingaDbCube
      */
-    abstract protected function getCube();
+    abstract protected function getCube(): IcingaDbCube;
 
     protected function moduleInit()
     {
         $this->cube = $this->getCube();
+        $this->cube->chooseFacts(array_keys($this->cube->getAvailableFactColumns()));
```

Comment:
Why is this now here?

Reply:
The fact columns have to be added to the cube anyway. So I added it here to simplify the code.